receiver/hostmetricsreceiver/internal/perfcounters/perfcounter_scraper.go

```diff
@@ -55,7 +63,8 @@ func (p *PerfLibScraper) Initialize(objects ...string) error {
 	for _, name := range objects {
 		index := nameTable.LookupIndex(name)
 		if index == 0 {
-			return fmt.Errorf("Failed to retrieve perf counter object %q", name)
+			p.logger.Error("Failed to retrieve perf counter object, performance data from this object will fail to be scraped", zap.String("counter", name))
```

Comment:
Shouldn't we log the error and don't attempt the scraping after that? I believe there is no way that the counter get available after the initialization, right? We can update the state here to reduce the number of objects to scrape in that case

Reply:
I'm trying to envision what this would look like. You're correct, once the counter is unavailable, it can't later become available.

I think I tended away from skipping scraping because there could potentially be multiple objects specified to `Initialize`, and it might make sense to "partially" scrape in those instances. But it looks like all instances where this struct is used, only use a single object here currently, so no failure here would end up being partial.

So maybe it makes sense to keep the error returned here, and just make scrape a no-op if `Initialize` errors due to the object not being available?
Thoughts on that approach?

Reply:
It is a good point that all occurrences of `Initialize` are only taking one object, but I think there's a solution that generalizes to either case (one or multiple objects):
- When _all_ objects fail to initialized, follow Dmitrii's suggestion to not scrape at all and update state
- When some (but not all), objects fail to initialize, scrape what we can and return partial error

Initialize could return a partial error when appropriate and this can be used to determine behavior of the scraper. WDYT?

Reply:
I'll work on implementing something similar to that.

Reply:
Changed the behavior to match the following:
* PerfLibScraper returns a partial error if any objects fail to initialize
  * objects that do not fail to initialize are still valid to be scraped, and are still set up appropriately
* For the current scrapers, any initialize error indicates a complete inability to scrape, so they skip their scrape step and do a no-op on scrape.
* On init, an error is printed if the Initialize function fails, indicating that the failing scraper will not scrape metrics.

Reply:
I think Dmitrii's point has been addressed so I'm resolving this.